harness/determined/experimental/_local_execution.py

```diff
@@ -0,0 +1,115 @@
+import contextlib
```

Comment:
non-blocking: why is this in `experimental`?  None of it will be user-facing, right?

Reply:
Yes. I want it to be together with `_native.py` since they are highly related. I also hate moving the functions to `experimental` folder just because they have a different namespace. I would import the functions in `determined` from `experimental`.